src/Join_SQL.php

```diff
@@ -119,9 +118,14 @@ public function initSelectQuery($model, $query)
                 isset($this->foreign_alias) ? (' '.$this->foreign_alias) : ''
             ),
             (
-                isset($this->owner->table_alias) ?
-                ($this->owner->table_alias.'.'.$this->master_field) :
-                ($this->owner->table).'.'.$this->master_field)
+                ($this->join
+                    // if join is nested, then use previous join table alias
+                    ? (isset($this->join->foreign_alias) ? $this->join->foreign_alias : $this->join->foreign_table)
+                    // otherwise use owner model table alias
+                    : (isset($this->owner->table_alias) ? $this->owner->table_alias : $this->owner->table)
+                ).'.'.$this->master_field
+            ),
+            $this->kind
         );
 
         if ($this->reverse) {
```

Comment:
there might be  a bit confusion. join is normally same as short_name. It was short_name originally. I think i started using ->join at some point or other way around but didn't complete. I think i will merge all of your PRs into single one, finish with joins add a bunch of test-scripts and then merge.